wordpress-mobile/WordPress-iOS
# What's New - Show announcements only when needed; + unit tests

Fixes #14984, #15016

Feature announcements should appear when users update the app, but should not appear if:

- users do a fresh install
- the corresponding release does not have published announcements

this PR also adds unit tests

To test:

**Test case 1**
- Fresh install and make sure that no feature announcements screen (see below) appears at startup
- go to Me/App settings and make sure that the table view does not contain a row named "What's New in WordPress" (at the moment, if present, it's the last row at the bottom of the table"
- Manually change the version of the `WordPress` target to 15.9, then relaunch
- Make sure that an announcements screen like the one below appears at startup
- go to Me/App settings, make sure that the row "What's New in WordPress"
- tap on that row, and make sure you see the same screen
- Manually change the version of the `WordPress` target to 15.10 (or anything greater than 15.9), then relaunch
- Make sure that no feature announcements screen (see below) appears at startup
- go to Me/App settings and make sure that the table view does not contain a row named "What's New in WordPress"

Example screen shot
<p align=center>
<img width=200 src=https://user-images.githubusercontent.com/34376330/94729397-356ee500-0327-11eb-8bd8-eaccc88ac54d.png>
</p>


**Test case 2**
- Run the `WordPressTest` test suite and make sure that all the tests succeed


PR submission checklist:

- [x] ~~I have considered adding accessibility improvements for my changes.~~ right on next PR.
